src/controllers/favorites.mjs

```diff
@@ -1,48 +1,35 @@
 import { getFileData, writeFile } from "../utils/fileHelper.mjs";
 import { CITIES_FILE_PATH } from "../../config.mjs";
 
-export const addCityToFavorites = async (req, res) => {
+export const toggleCityFavorite = async (req, res) => {
   const cityName = req.params.cityName.toLowerCase();
+  const addToFavorites = req.body.addToFavorites;
   try {
     const cities = await getFileData(CITIES_FILE_PATH);
     const cityIndex = cities.findIndex(
       (city) => city.name.toLowerCase() === cityName
     );
+
     if (cityIndex === -1) {
       return res.status(400).json({ message: "City not found" });
     }
-    cities[cityIndex].isFavorite = true;
-    await writeFile(CITIES_FILE_PATH, cities);
-    res
-      .status(200)
-      .json({ message: `City ${cityName} added to favorites successfully` });
-  } catch (err) {
-    console.log(err);
-    res.status(500).json({ message: "Error adding city to favorites" });
-  }
-};
 
-export const deleteCityFromFavorites = async (req, res) => {
-  const cityName = req.params.cityName.toLowerCase();
-
-  try {
-    const cities = await getFileData(CITIES_FILE_PATH);
-    const cityIndex = cities.findIndex(
-      (city) => city.name.toLowerCase() === cityName
-    );
-
-    if (cityIndex === -1) {
-      return res.status(400).json({ message: "City not found" });
+    if (addToFavorites === "true") {
+      cities[cityIndex].isFavorite = true;
+      res
+        .status(200)
+        .json({ message: `City ${cityName} added to favorites successfully` });
+    } else {
+      delete cities[cityIndex].isFavorite;
```

Comment:
No, forget about `delete`, super bad practice
I would propose you to investigate one new array build-in feature - `array.splice`
This way you would be able to replace an item inside an array

Pay attention, that this method is changing variable itself, keep this in mind

Reply:
Or you can try with `cities[cityIndex].isFavorite = undefined`